### PR TITLE
Added Agitator Subsystem

### LINKED ISSUE #64: Agitator Subsystem
Production software for the agitator subsystem in the robot code, including:
- interface definitions
- commands
- PID if applicable
- tested on the robot to make sure this does not break anything
- merged and integrated with 2020 robot code in github
- ready for integration

### PR DESCRIPTION
**Pull request by aadiamazing5**
Added agitator subsystem, with 3 simple commands. Should eventually be integrated with intake subsystem. Will be bulletproofed soon.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/AgitatorSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+public void reverseAgitator ()
+{
+    agitatorMotor.set(AGITATOR_REVERSE_SPEED);
+    } //Runs the agitator in reverse (Might be useless?)
+
+
+//Variables:
+private final double AGITATOR_SPEED = 1.0;
+private final double AGITATOR_REVERSE_SPEED = -1.0;
```
Comment (kevlam2706) [inline src/main/java/frc/robot/subsystems/AgitatorSubsystem.java]: Typical convention is to define these up top so we're not left in suspense as to what the constants are...

File: src/main/java/frc/robot/subsystems/AgitatorSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.config.Config;
+import frc.robot.config.FluidConstant;
+
+public class Agitator extends SubsystemBase {
+
+    private VictorSPX agitatorMotor;
+
+
+public Agitator (VictorSPX agitatorMotor)
```
Comment (kevlam2706) [inline src/main/java/frc/robot/subsystems/AgitatorSubsystem.java:17]: Couple of things here - one, all other subsystems were defined using a singleton pattern - can you have a look at one of the existing ones and follow that pattern?

Also, your constructor assumes someone else will instantiate the VictorSPX motor controller object and I think the convention we've used elsewhere is for the subsystems to each instantiate and manage their own. You'll also need to know the CAN ID of the Victor, which you'll pull from the robot constants.

**Review by kevlam2706 — APPROVED** [review]
You can merge this and continue refining it if you like, but I'd like to see you refactor this to use a singleton pattern. You should also tie it into the main robot code and you can test it by temporarily binding a joystick command to run the agitator. (See existing code for hints on how to do that)

You can test this code by deploying it on Mergonaut and setting your agitator motor to use the CAN ID of Mergonaut's intake motor, which was also controlled by a Victor SPX.

File: src/main/java/frc/robot/subsystems/AgitatorSubsystem.java (hunk 2 above)
Comment (aadiamazing5) [reply src/main/java/frc/robot/subsystems/AgitatorSubsystem.java:17]: will do, ill get the changes done by wednesday, then test it then

Comment (nicojopr) [reply src/main/java/frc/robot/subsystems/AgitatorSubsystem.java:17]: Here is a reference to the preferred Singleton pattern 
https://github.com/georgetzavelas/guava-examples/blob/master/src/com/design/pattern/creational/singleton/Singleton.java

File: src/main/java/frc/robot/subsystems/AgitatorSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.config.Config;
+import frc.robot.config.FluidConstant;
+
+public class Agitator extends SubsystemBase {
+
+    private VictorSPX agitatorMotor;
+
+//Variables:
+private final double AGITATOR_SPEED = 1.0;
```
Comment (Zinka010) [inline src/main/java/frc/robot/subsystems/AgitatorSubsystem.java]: As a general rule, you should indent your code properly using 4 space indentation. I believe there is a VS Code plugin called Beautify File that could help you do this, though I don't know how well it works with java files.

File: src/main/java/frc/robot/subsystems/AgitatorSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    agitatorMotor.set(AGITATOR_SPEED);
+    } 
+
+public void stopAgitator ()
+{ // Stops the agitator
+    agitatorMotor.set(0.0);
+    }
+
+public void reverseAgitator ()
+{//Runs the agitator in reverse (Might be useless?)
```
Comment (Zinka010) [inline src/main/java/frc/robot/subsystems/AgitatorSubsystem.java]: Remove the `(Might be useless?)`, worst case it just isn't used

**Review by Zinka010 — APPROVED** [review]
All the elements are there, you just need to format properly and fix a few other nitpicks. And of course, make it a Singleton as @kevlam2706 said

**Review by Zinka010 — CHANGES_REQUESTED** [review]
See previously requested changes

**Review by Zinka010 — APPROVED** [review]
Alright so most of the pieces are there and with the competition looming I'm just going to quickly adapt it to fit the rest of the codebase. I'll do this on Friday so if you're there we can go through it together on your laptop but otherwise we just need it to be done.

**Comment by aadiamazing5** [comment]
@Zinka010 So I've gone through and changed the file on my computer to a singleton pattern (likely won't commit because I'm having trouble getting it to work). I unfortunately won't be able to make it on Friday due to a badminton tournament, but please let me know what tweaks you decide to make to the file so that I can learn and then update my code accordingly. Thanks :)

**Comment by aadiamazing5** [comment]
@Zinka010 Would you like me to merge as it is for now or wait until Friday?

**Comment by kevlam2706** [comment]
If it isn't working then you shouldn't merge, but maybe Justin can
take what you have, make it work, and them commit it into master.

Then you can either pull from master to get it back into your branch,
or just close your branch without merging.

**Comment by aadiamazing5** [comment]
It works as it is right now on this branch, it just doesn't work on my local device where I was tinkering with Singletons. Merging this branch shouldn't cause any problems. Let me know what you'd like me to do

**Pull request merged**